### ISSUE TITLE
(mb_)strlen misused inspection suggests incorrect substitution on null value.

### ISSUE THREAD

**Issue by WiXSL** [kalessil/phpinspectionsea#1142]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive                          |
| Plugin         | Php Inspections (EA Extended) Latest |
| Language level | PHP 7.2                         |

Current behaviour (description/screenshot:)
```
        $foo = null;

        // Here, the plugin suggests to use  $foo !== '' instead of strlen($foo) > 0
        // But strlen($foo) > 0 evaluates as false while $foo !== '' evaluates as true

        if (strlen($foo) > 0) {
            return 'OK';
        }

        if ($foo !== '') {
            return 'OK';
        }
```

Environment details
---
Information from Help -> About dialog (contains IDE and it's version details).

PhpStorm 2018.3.2
Build #PS-183.4886.22, built on December 12, 2018
Licensed to Anibal Svarcas
Subscription is active until August 1, 2019
JRE: 1.8.0_152-release-1343-b26 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

**Comment by kalessil** [comment]
Will check, perhaps we can fix this by making types analysis more precise.

**Comment by kalessil** [comment]
Fixed!

**Comment by PhilippeThouvenot** [comment]
Hi, same problem with 3.0.13 version.

**Comment by kalessil** [comment]
Hi @PhilippeThouvenot, would you please file a new issue then with code sample?
